gpx.go

```diff
@@ -17,15 +17,17 @@ import (
 
 /*==========================================================*/
 
-const timelayout = "2006-01-02T15:04:05Z"
 const defaultStoppedSpeedThreshold = 1.0
 
 /*==========================================================*/
 
+// Waypoints is a collection of waypoints whether in a track, a route, or standalone.
+type Waypoints []Wpt
+
 // Trkseg is a GPX track segment
 type Trkseg struct {
-	XMLName xml.Name `xml:"trkseg"`
-	Points  []Wpt    `xml:"trkpt"`
+	XMLName   xml.Name `xml:"trkseg"`
+	Waypoints `xml:"trkpt"`
```

Comment:
By [embedding](https://golang.org/doc/effective_go.html#embedding) Waypoints, all its methods get promoted to Trkseg. So we can say Gpx.Tracks[0].Segments[0].Length2D() instead of Gpx.Tracks[0].Segments[0].Points.Length2D().